src/logic/withEnv.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+import React from "react";
+import { changes, getAllEnvs } from "@ledgerhq/live-common/lib/env";
+import type { EnvName } from "@ledgerhq/live-common/lib/env";
+
+const withEnv = (name: EnvName, propName: string = "env") => (Comp: any) =>
+  class WithEnvs extends React.Component<*, { env: any }> {
+    state = {
+      env: null,
+    };
+
+    sub: *;
+
+    componentDidMount() {
+      this.init();
+      this.subscribe();
+    }
+
+    componentWillUnmount() {
+      if (this.sub) {
+        this.sub.unsubscribe();
+      }
+    }
+
+    init = () => {
+      const envs = getAllEnvs();
+      this.setState({ env: envs[name] });
```

Comment:
we shouldn't do a setState at init, it should directly be in the state initially, this would avoid a double rendering.